agent/utils/uuidwrapper/uuid.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package uuidwrapper
+
+import (
+	"github.com/pborman/uuid"
+)
+
+// UUID wraps 'uuid' methods for testing
+type UUID interface {
```

Comment:
Call this UUIDProvider or something that implies that it generates a UUID.

Otherwise, it sounds like the object itself *is* a uuid. :)